internal/admission/validation/gateway/httproute.go

```diff
@@ -97,22 +98,69 @@ func validateHTTPRouteListener(listener *gatewayapi.Listener) error {
 // HTTPRoute implementation and validates that the provided object is not using
 // any of those unsupported features.
 func validateHTTPRouteFeatures(httproute *gatewayapi.HTTPRoute, translatorFeatures translator.FeatureFlags) error {
-	for _, rule := range httproute.Spec.Rules {
-		for _, match := range rule.Matches {
-			// We support query parameters matching rules only with expression router.
-			if len(match.QueryParams) != 0 {
-				if !translatorFeatures.ExpressionRoutes {
-					return fmt.Errorf("queryparam matching is supported with expression router only")
-				}
+	unsupportedHTTPRouteFilterTypes := []gatewayapi.HTTPRouteFilterType{
+		gatewayapi.HTTPRouteFilterURLRewrite,
+		gatewayapi.HTTPRouteFilterRequestMirror,
+	}
+	unsupportedFilterMap := lo.SliceToMap(unsupportedHTTPRouteFilterTypes, func(t gatewayapi.HTTPRouteFilterType) (gatewayapi.HTTPRouteFilterType, struct{}) {
+		return t, struct{}{}
+	})
+
+	const (
+		KindService = gatewayapi.Kind("Service")
+		KindGateway = gatewayapi.Kind("Gateway")
+	)
+
+	// Validate parentRefs: only allow empty, `/Gateway` or `gateway.networking.k8s.io/Gateway`
+	// REVIEW: Should this happen before we get here? we search for parent gateways before we reach here.
```

Comment:
Yes, I think we should have a proper function or something to check `ParentRefs` before validating features. I think that supporting only Gateways as Parents should not be seen (and validated) as a feature.

Reply:
extracted it out and run it before validating features.